src/audio/module_adapter/module/modules.c

```diff
@@ -161,7 +163,8 @@ static int modules_init(struct processing_module *mod)
 
 	/* Call module specific init function if exists. */
 	if (mod->is_native_sof) {
-		const struct module_interface *mod_in = md->ops;
+		struct module_interface *mod_in =
+					(struct module_interface *)md->module_adapter;
 
```

Comment:
That changes break loadable modules https://github.com/thesofproject/sof/pull/8667 which works on main. Please do not introduce changes breaking working flow.

Reply:
@pjdobrowolski I don't see how this revert can break anything. As you see from the hunk above around lines 90-110 the version before this commit stored a pointer to native loadable module interface, returned by `native_system_agent_start()` in `md->ops` and after this commit it's in `md->module_adapter`. The effect of this is that before this commit all framework calls to interface functions went directly to the loaded module functions, whereas after this commit they first call wrappers from module.c, which then call respective methods from the loaded module.
The difference is particularly visible for `.free()` and `.init()`. For the `.free()` case you have to call the wrapper, otherwise you miss calling `lib_manager_free_module()`. As for `.init()` this change probably has the effect, that if you don't reload the loaded module, then the second call to `.init()` would go directly to the loaded module, instead of `modules_init()`. Is this what is breaking your tests?